DRILL-6436: Storage Plugin to have name and context moved to Abstract…

ilooner opened the pull request:
…StoragePlugin

ilooner commented:
@amansinha100 please review.

ilooner commented:
Travis failure is unrelated. I have seen Travis timeout occasionally on other branches. I have create a jira to investigate this https://issues.apache.org/jira/browse/DRILL-6439 .

paul-rogers commented on exec/java-exec/src/main/java/org/apache/drill/exec/store/AbstractStoragePlugin.java:
```diff
@@ -29,13 +29,20 @@
 
-  protected AbstractStoragePlugin() { }
+  protected final DrillbitContext context;
+  final String name;
```
`protected`. Actually, since you've provided `getFoo()` methods, these can be private.

ilooner commented:
@vrozov @paul-rogers 

Made context and name private final fields. Also removed redundant getContext() methods. And fixed javadoc errors in AbstractStoragePlugin.
